### PR TITLE
GEOMESA-1795 Move density iterator result to user data

### PR DESCRIPTION
**Pull request by aheyne**
Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala [hunk 1]
```diff
@@ -145,6 +146,7 @@ object KryoLazyDensityIterator extends LazyLogging {
   lazy val DENSITY_SFT = SimpleFeatureTypes.createType("density", "result:String,*geom:Point:srid=4326")
 
   val DEFAULT_PRIORITY = 25
+  val DENSITY_VALUE    = new ClassKey(classOf[java.lang.String])
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala]: shouldn't this be `classOf[Array[Byte]]`?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala [hunk 2]
```diff
@@ -227,7 +229,7 @@ object KryoLazyDensityIterator extends LazyLogging {
    * Decodes a result feature into an iterator of (x, y, weight)
    */
   private def decodeResult(gridSnap: GridSnap)(sf: SimpleFeature): Iterator[(Double, Double, Double)] = {
-    val result = sf.getAttribute(0).asInstanceOf[Array[Byte]]
+    val result = sf.getUserData.get(DENSITY_VALUE).asInstanceOf[Array[Byte]]
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala]: maybe we should add a check for the attribute as a fallback to preserve version compatibility

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala (hunk 1 above)
Comment (aheyne) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala]: Yeah, fixed

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala [hunk 3]
```diff
@@ -145,6 +146,7 @@ object KryoLazyDensityIterator extends LazyLogging {
   lazy val DENSITY_SFT = SimpleFeatureTypes.createType("density", "result:String,*geom:Point:srid=4326")
 
   val DEFAULT_PRIORITY = 25
+  val DENSITY_VALUE    = new ClassKey(classOf[Byte])
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyDensityIterator.scala]: oops, still wrong :P - should be `Array[Byte]`

**Comment by elahrvivaz** [comment]
i fixed the classkey but it still has conflicts with master

**Comment by aheyne** [comment]
Yeah, #1478 broke this by moving the code to `KyroLazyDensityUtils`. This will have to be redone.

**Comment by aheyne** [comment]
Fixed, should be good to go.

**Pull request merged**